python/cudf/cudf/tests/test_spilling.py

```diff
@@ -254,6 +254,7 @@ def reload_options():
     manager = get_global_manager()
     assert isinstance(manager, SpillManager)
     assert manager.statistics.level == 2
+    monkeypatch.delenv("CUDF_SPILL")
```

Comment:
I think this fix is on the right track but insufficient. I think you also need to call `reload_options` after this so that the actual option is reset.

Reply:
`monkeypatch.reload_options`?

Reply:
Nope it's defined as a helper within this test itself, see the top of the test function.

Reply:
For completeness I would also unset all the other variables (CUDF_SPILL_ON_DEMAND, CUDF_SPILL_DEVICE_LIMIT, CUDF_SPILL_STATS) too just to avoid any other state persisting.

Reply:
Done.